sale_order_line_sequence/model/sale_order.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2017 Eficent Business and IT Consulting Services S.L.
+# Copyright 2017 Serpent Consulting Services Pvt. Ltd.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    @api.multi
+    @api.depends('order_line')
+    def _compute_max_line_sequence(self):
+        """Allow to know the highest sequence entered in sale order lines.
+        Then we add 1 to this value for the next sequence.
+        This value is given to the context of the o2m field in the view.
+        So when we create new sale order lines, the sequence is automatically
+        added as :  max_sequence + 1
+        """
+        for sale in self:
+            sale.max_line_sequence = (
+                max(sale.mapped('order_line.sequence') or [0]) + 1)
+
+    max_line_sequence = fields.Integer(
+        string='Max sequence in lines',
+        compute='_compute_max_line_sequence',
+        store=True
+    )
+
+    @api.multi
+    def _reset_sequence(self):
+        for rec in self:
+            current_sequence = 1
+            for line in rec.order_line:
+                line.sequence = current_sequence
+                current_sequence += 1
+
+    @api.multi
+    def write(self, line_values):
+        res = super(SaleOrder, self).write(line_values)
+        self._reset_sequence()
+        return res
+
+    @api.multi
+    def copy(self, default=None):
+        return super(SaleOrder,
+                     self.with_context(keep_line_sequence=True)).copy(default)
+
+
+class SaleOrderLine(models.Model):
+    _inherit = 'sale.order.line'
+
+    # re-defines the field to change the default
+    sequence = fields.Integer(
+        help="Gives the sequence of this line when displaying the sale order.",
+        default=9999,
+        string="Sequence"
+    )
+
+    # displays sequence on the order line
+    sequence2 = fields.Integer(
+        help="Shows the sequence of this line in the sale order.",
+        related='sequence',
+        string="Sequence 2",
```

Comment:
Can you please change this string to something less ugly? This appears in the SO view.

Reply:
@aheficent Yes I know, But if name is same then runbot get warning like this https://github.com/OCA/sale-workflow/pull/744#issuecomment-441557570

That's why I put name `Sequence 2`.

Reply:
Can you put in the sequence field then `string="Hidden sequence"` ? Then you will be able to put sequence in the other

Reply:
@aheficent Done

Reply:
Sorry I didn't explain it correctly. I meant to change the string of the sequence field. But I realize now it was also a bad idea.

I think is better what you did but with a more intuitive string. Can you put something like:

```
     sequence2 = fields.Integer(
        help="Shows the sequence of this line in the sale order.",
        related='sequence',
        string="Line Number",
        readonly=True,
        store=True
    )
```